.github/workflows/coverity-scan.yaml

```
@@ -0,0 +1,45 @@
+# Tests in this workflow are run periodically.
+# They are generally very expensive and not on the critical path
+# for development
+# scheduled workflows only run in the default branch
+
+name: Coverity Scan
+
+on:
+  schedule:
+    - cron: "42 3 * * 0" # once a week
+  # allows to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  coverity-scan:
+    name: Scan with Coverity
+    # only run the workflow on Squid's main repository
+    if: github.repository == 'squid-cache/squid'
+
+    runs-on: ubuntu-22.04
+    # this job only runs if the github repository contains username
+    # and password to access the Coverity Scan service
+    env:
+      coverity_user: ${{ secrets.COVERITY_USER }}
+      coverity_token: ${{ secrets.COVERITY_TOKEN }}
+
+    container:
+      image: squidcache/buildfarm-coverity:stable
+      options: --user 1001 # uid used by worfklow runner
+
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v4
+
+      - name: Scan
+        run: |
+          cov-build --dir cov-int ./test-builds.sh layer-02-maximus
+          tar caf cov-int.tar.xz cov-int
```

For several reasons, I recommend avoiding "traditional" (a.k.a. "bundled”) option style, but I do not insist on this change, and do not want to detail the reasons behind it:

```suggestion
          tar -caf cov-int.tar.xz cov-int
```